mathesar/views/api.py

```diff
@@ -77,6 +83,62 @@ def type_suggestions(self, request, pk=None):
         return Response(col_types)
 
 
+class ColumnViewSet(viewsets.ViewSet):
+    queryset = Table.objects.all().order_by('-created_at')
+
+    def list(self, request, table_pk=None):
+        paginator = ColumnLimitOffsetPagination()
+        columns = paginator.paginate_queryset(self.queryset, request, table_pk)
+        serializer = ColumnSerializer(columns, many=True)
+        return paginator.get_paginated_response(serializer.data)
+
+    def retrieve(self, request, pk=None, table_pk=None):
+        table = Table.objects.get(id=table_pk)
+        try:
+            column = table.sa_columns[int(pk)]
+        except IndexError:
+            raise NotFound
+        serializer = ColumnSerializer(column)
+        return Response(serializer.data)
+
+    def create(self, request, table_pk=None):
+        table = Table.objects.get(id=table_pk)
+        # We only support adding a single column through the API.
+        serializer = ColumnSerializer(data=request.data, context={'request': request})
+        if serializer.is_valid():
+            try:
+                column = table.add_column(request.data)
+                out_serializer = ColumnSerializer(column)
+                return Response(out_serializer.data, status=status.HTTP_201_CREATED)
+            except ProgrammingError as e:
+                if type(e.orig) == DuplicateColumn:
+                    raise ValidationError(
+                        f"Column {request.data['name']} already exists"
+                    )
+                else:
+                    raise APIException(e)
+        else:
+            raise ValidationError(serializer.errors)
+
+    def partial_update(self, request, pk=None, table_pk=None):
+        table = Table.objects.get(id=table_pk)
+        assert isinstance((request.data), dict)
+        try:
+            column = table.alter_column(pk, request.data)
+        except ProgrammingError as e:
+            if type(e.orig) == UndefinedFunction:
+                raise ValidationError("This type cast is not implemented")
+            else:
+                raise ValidationError
```

Comment:
Should we also catch an `IndexError` here, like in `retrieve()`?

Reply:
Good idea, will do.